crypto/src/main/java/net/corda/v5/crypto/DigestAlgorithmName.java

```diff
@@ -1,53 +1,70 @@
 package net.corda.v5.crypto;
 
 import java.util.Locale;
+
 import net.corda.v5.base.annotations.CordaSerializable;
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 
 /**
  * The digest algorithm name. This class is to be used in Corda hashing API.
- *
- * @param name The name of the digest algorithm to be used for the instance.
  */
 
 @CordaSerializable
 public final class DigestAlgorithmName {
-    @NotNull private final String name;
+    /**
+     * Construt a digest algorith name.
+     * 
+     *  * @param name The name of the digest algorithm to be used for the instance.
+     */
+    public DigestAlgorithmName(@NotNull String name) {
+        super();
+        this.name = name;
+        if (this.name.isBlank()) throw new IllegalArgumentException("Hash algorithm name unavailable or not specified");
+    }
+
+    @NotNull
+    private final String name;
```

Comment:
Fine for constructor to go up here. Can we please revert `name` property to be on top please.

Reply:
done